AAEmu.Game/Models/Game/Items/Loots/LootPack.cs

```diff
@@ -222,11 +222,11 @@ public class LootPack
                 }
 
                 // Group 0 items will always need to be included
-                if (loot.Group <= 0)
+                if (loot.Group <= 0 || loot.AlwaysDrop || loot.DropRate == 10000000)
                 {
-                    if (!selectedItemsByGroup.ContainsKey(loot.Group))
-                        selectedItemsByGroup.Add(loot.Group, []);
-                    selectedItemsByGroup[loot.Group].Add(loot);
+                    if (!selectedItemsByGroup.ContainsKey(0))// replaced loot.Group==0 with 0 to make it easier to see.
+                        selectedItemsByGroup.Add(0, []);
+                    selectedItemsByGroup[0].Add(loot);
```

Comment:
Not related to the correctness of the code, but you can rewrite it to be more efficient (1 dictionary access vs 3 with the original):
```suggestion
                    ref var lootList = ref CollectionsMarshal.GetValueRefOrAddDefault(selectedItemsByGroup, 0u, out var exists); // replaced loot.Group==0 with 0 to make it easier to see.
                    if (!exists)
                        lootList = [];
                    lootList.Add(loot);
```